datasets/simulated_block.py

```diff
@@ -14,55 +13,41 @@ class Dataset(BaseDataset):
 
     # List of parameters to generate the datasets. The benchmark will consider
     # the cross product for each key in the dictionary.
-    # cos + bruit ~ N(mu, sigma)
+    # blocked signal + bruit ~ N(mu, sigma)
     parameters = {
         'sigma': [0.1],
         'mu': [0],
-        'K': [500],
-        'len_output': [400],
+        'n_features': [50],
+        'n_samples': [40],
         'type_A': ['identity', 'random_square', 'random_nonsquare'],
-        'num_block': [10]}
+        'num_block': [3]}
 
-    def __init__(self, mu=0, sigma=0.3, K=10, len_output=5,
+    def __init__(self, mu=0, sigma=0.3, n_features=10, n_samples=5,
                  type_A='identity', random_state=27, num_block=1):
         # Store the parameters of the dataset
         self.mu = mu
         self.sigma = sigma
-        self.K, self.len_output = K, len_output
+        self.n_features, self.n_samples = n_features, n_samples
         self.type_A = type_A
         self.num_block = num_block
         self.random_state = random_state
 
     def set_A(self, rng):
         if self.type_A == 'random_square':
-            A = rng.randn(self.K, self.K)
+            A = rng.randn(self.n_features, self.n_features)
         elif self.type_A == 'random_nonsquare':
-            A = rng.randn(self.K, self.len_output)
+            A = rng.randn(self.n_features, self.n_samples)
         else:
-            A = np.identity(self.K)
+            A = np.identity(self.n_features)
         return A
 
     def get_data(self):
         rng = np.random.RandomState(47)
-        random.seed(47)
-        slice_list = sorted(random.sample(range(self.K),
-                                          self.num_block - 1),
-                            reverse=False)
-        slice_list.insert(0, 0)
-        slice_list.append(self.K)
-        slice_schedule = [np.ones(self.K).tolist()[slice_list[i]:
-                                                   slice_list[i + 1]]
-                          for i in range(self.num_block)]
-
-        rvs = stats.norm(loc=0, scale=2).rvs
-        slice_value = sprandom(1, self.num_block, density=0.5,
-                               random_state=rng, data_rvs=rvs).A[0]
-        y = np.array([])
-        for i in range(self.num_block):
-            y = np.concatenate(
-                (y, np.array(slice_schedule[i]) * slice_value[i]))
-
-        y = np.array(y) + rng.normal(self.mu, self.sigma, self.K)
+        rvs = stats.norm(loc=0, scale=1).rvs
+        y = np.cumsum(sprandom(1, self.n_features,
+                      density=self.num_block/self.n_features,
+                      random_state=rng, data_rvs=rvs).A[0]) + \
+            rng.normal(self.mu, self.sigma, self.n_features)
```

Comment:
if you write $y = X w$ it's $w$ that should be the block signal not $y$

Reply:
In our context, the question is $ .5 * \|| y - A x \||^2 $. Until now I set $y$ as noisy signal, but not operater * noisy signal. So here why should $w$ be the block signal?

Reply:
ok then if you write

$y = Ax + n$ then $x$ should be the block signal and $n$ is the added noise.